include/circt/Dialect/FIRRTL/FIRRTLExpressions.td

```diff
@@ -412,6 +413,16 @@ let inferType = "impl::inferBitwiseResult" in {
   }
 }
 
+let inferType = "impl::inferElementwiseResult" in
+let hasFolder = 0 in {
+  def ElementwiseAndPrimOp : BinaryPrimOp<"elementwise_and", PassiveType,
+                                PassiveType, PassiveType, [Commutative]>;
+  def ElementwiseOrPrimOp  : BinaryPrimOp<"elementwise_or", PassiveType,
+                                PassiveType, PassiveType, [Commutative]>;
+  def ElementwiseXorPrimOp : BinaryPrimOp<"elementwise_xor", PassiveType,
+                                PassiveType, PassiveType, [Commutative]>;
+}
+
```

Comment:
You probably want to make sure that the element type is not Analog, Reset, or Bundle.
https://github.com/llvm/circt/blob/main/include/circt/Dialect/FIRRTL/FIRRTLExpressions.td#L409

If this operation is allowed on nested vectors, are they handled correctly everywhere?

Reply:
Thanks, rejecting Analog and Reset makes sense to me. I'll add more strict checker. I think technically we can allow bundles (though there won't be immediate users) so it's open question. 

> If this operation is allowed on nested vectors, are they handled correctly everywhere?

Yes, I think so. If we want to generate these operations from frontends, it would be necessary to support inferwidth but rest of the passes should be working without a problem.

Reply:
Restricted to UInt1Dvector to simplify the things.